app-modules/engagement/database/migrations/2025_04_17_142824_data_migration_refactor_engagement_subject_to_jsonb.php

```diff
@@ -0,0 +1,224 @@
+
+    For more information or inquiries please visit our website at
+    https://www.canyongbs.com or contact us via email at legal@canyongbs.com.
+
+</COPYRIGHT>
+*/
+
+use App\Features\RefactorEngagementCampaignSubjectToJsonb;
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Support\Facades\DB;
+
+return new class () extends Migration {
+    public function up(): void
+    {
+        DB::transaction(function () {
+            DB::table('engagements')->select('id', 'subject')->chunkById(500, function ($rows) {
+                foreach ($rows as $row) {
+                    $json = [
+                        'type' => 'doc',
+                        'content' => [[
+                            'type' => 'paragraph',
+                            'attrs' => ['class' => null, 'style' => null],
+                            'content' => [[
+                                'type' => 'text',
+                                'text' => $row->subject,
+                            ]],
+                        ]],
+                    ];
+                    DB::table('engagements')
+                        ->where('id', $row->id)
+                        ->update(['subject' => json_encode($json)]);
+                }
+            });
+
+            DB::table('engagement_batches')->select('id', 'subject')->chunkById(500, function ($rows) {
+                foreach ($rows as $row) {
+                    $json = [
+                        'type' => 'doc',
+                        'content' => [[
+                            'type' => 'paragraph',
+                            'attrs' => ['class' => null, 'style' => null],
+                            'content' => [[
+                                'type' => 'text',
+                                'text' => $row->subject,
+                            ]],
+                        ]],
+                    ];
+                    DB::table('engagement_batches')
+                        ->where('id', $row->id)
+                        ->update(['subject' => json_encode($json)]);
+                }
+            });
+
+            DB::table('form_email_auto_replies')->select('id', 'subject')->chunkById(500, function ($rows) {
+                foreach ($rows as $row) {
+                    $json = [
+                        'type' => 'doc',
+                        'content' => [[
+                            'type' => 'paragraph',
+                            'attrs' => ['class' => null, 'style' => null],
+                            'content' => [[
+                                'type' => 'text',
+                                'text' => $row->subject,
+                            ]],
+                        ]],
+                    ];
+                    DB::table('form_email_auto_replies')
+                        ->where('id', $row->id)
+                        ->update(['subject' => json_encode($json)]);
+                }
+            });
+
+            DB::table('campaign_actions')
```

Comment:
Can we use a json where query here to specify this should only be for `campaign_actions` that are for engagements?